### PR TITLE
refactor heartbeat between worker and log_server

### PR DESCRIPTION
**Pull request by zenghsh3**

### CODE REVIEW

File: parl/remote/log_server.py [hunk 1]
```diff
@@ -84,6 +88,24 @@ def download_log():
...
+
+    try:
+        socket.send_multipart([
+            remote_constants.HEARTBEAT_TAG,
+            to_byte(heartbeat_server_addr),
+        ])
+        message = socket.recv_multipart()
+    except zmq.error.Again as e:
+        err_str = "Can not connect to the worker please " \
+                  "check if the worker is started and ensure the input " \
```
Comment (TomorrowIsAnOtherDay) [inline parl/remote/log_server.py]: check if the worker is running.

File: parl/remote/log_server.py [hunk 2]
```diff
@@ -84,6 +88,24 @@ def download_log():
         return send_file(log_file_path, as_attachment=True)
 
 
+def send_heartbeat_addr_to_worker(worker_addr, heartbeat_server_addr):
+    ctx = zmq.Context()
+    socket = create_client_socket(ctx, worker_addr, heartbeat_timeout=True)
+
+    try:
+        socket.send_multipart([
+            remote_constants.HEARTBEAT_TAG,
```
Comment (TomorrowIsAnOtherDay) [inline parl/remote/log_server.py]: just send the `NORMAL_TAG`.

Comment (zenghsh3) [reply parl/remote/log_server.py]: done

File: parl/remote/log_server.py (hunk 1 above)
Comment (zenghsh3) [reply parl/remote/log_server.py]: done

**Review by TomorrowIsAnOtherDay — APPROVED** [review]
LGTM.

**Pull request merged**